xenserver/provision.erb

```diff
@@ -0,0 +1,32 @@
+<%#
+kind: provision
+name: XenServer default answerfile
+oses:
+- XenServer
+%>
+<% subnet = @host.subnet -%>
+<% dhcp = subnet.dhcp_boot_mode? -%>
+<% interface_name = @host.primary_interface.identifier.empty? ? 'eth0' : @host.primary_interface.identifier -%>
```

Comment:
Since 1.8+ it should return interface object so the snippet would return identifier of primary interface. Before 1.8 `@host.primary_interface` would return identifier itself. So yes, looks good to me, I'd just prefer to use `blank?` instead of `empty?`. When creating host from UI it could be empty string, from API it can be nil.